Update cmake scripts to build a .dmg

srcejon opened the pull request:
This PR updates the cmake scripts so that a .dmg can be built for easy installation on MacOS - hopefully without needing to install anything else.

Rather than use cpack's fixup_bundle, it now uses macdeployqt, so that all the required Qt stuff is included.

It also updates the external/CMakeLists.txt to build nearly all of the required 3rd party libraries, from source, so doesn't require mac ports. I've updated the build instructions on the wiki.

There's also a small fix so that the 3D Maps work.

f4exb commented:
It makes the MSVC build fail

f4exb commented:
Ubuntu build fails: https://ci.appveyor.com/project/f4exb/sdrangel/builds/43581106/job/wswv9uwrv70dk1y7#L8362
```
   dh_auto_build -O--buildsystem=cmake\+ninja
	cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 ninja -j8 -v
ninja: error: 'external/opus/lib/libopus.so', needed by 'lib/libsdrbase.so', missing and no known rule to make it
dh_auto_build: error: cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 ninja -j8 -v returned exit code 1
```
`libopus-dev` is a system wide installed package and a dependency:
  - https://github.com/f4exb/sdrangel/blob/master/.appveyor.yml#L90
  - https://github.com/f4exb/sdrangel/blob/master/debian/control#L48

srcejon commented:
> Hello, Where can I find instructions to build that DMG for MacOs? Thanks.

https://github.com/f4exb/sdrangel/wiki/Compile-in-MacOS

A v7 DMG is here: http://sdrangel.org/downloads/sdrangel-7.0.0-macos.dmg

loewal commented:
Thank you very much Jon,
I prefer to stay with the compilation I did according Linux-instructions. Works well!
But on the other hand I wonder why the DMG is not part of the artifacts like Windows etc....
Can I replace th "7.0.0" etc.. with "7.0.1" in the future, or is there a folder where I can download from?

Thnx anyway!!

srcejon commented:
I still need to setup automatic builds for it (and I've just noticed there is a problem with the current script). Still a work in progress...

srcejon commented:
No - but I've updated the wiki - thanks.

I broke the Mac build when I fixed the Ubuntu build - fix is here: https://github.com/f4exb/sdrangel/compare/master...srcejon:mac_package

I'm trying now to see if I can get it to build on appveyor. May need some changes to build in under 60 minutes.

srcejon commented:
Takes far too long to build to do via appveyor. However, I can setup a simple build on a Mac VM I have.

Using the github CLI tools (https://github.com/cli/cli/releases), I can just poll for a new SDRangel release tag with:

    gh release view --json tagName -q .tagName > tag

Then kick off a build, and upload the .dmg as an asset to the release page with:

    gh release upload ${tag} sdrangel-*.dmg

@f4exb - are you OK with that? I think as a Collaborator I have the permissions to do it.

loewal commented:
Would be great!!
On 21 May 2022, 19:23 +0200, srcejon ***@***.***>, wrote:
> Takes far too long to build to do via appveyor. However, I can setup a simple build on a Mac VM I have.
> Using the github CLI tools (https://github.com/cli/cli/releases), I can just poll for a new SDRangel release tag with:
> gh release view --json tagName -q .tagName > tag
> Then kick off a build, and upload the .dmg as an asset to the release page with:
> gh release upload ${tag} sdrangel-*.dmg
> @f4exb - are you OK with that? I think as a Collaborator I have the permissions to do it.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.Message ID: ***@***.***>

f4exb commented:
@srcejon yes I think you are able to do upload. I know Appveyor may be problematic but I think Github actions also have a Mac VM for builds. The nice think about automatic CI/CD is that it is automatic precisely. That would save you the pain of having to follow up the releases and also may offer intermediate checkpoints.

srcejon commented:
> @srcejon yes I think you are able to do upload.

Yep, thanks - there are .dmgs as assests on the release page now.

> I know Appveyor may be problematic but I think Github actions also have a Mac VM for builds. 

Github doesn't appear to have a Mojave VM - which is what someone requested. (According to https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)

Can possibly speed up Appveyor builds, by building the libraries as a separate project (like Windows) - but that will take some more time to set up. Will come back to it at some point, but a few other bugs to try to sort out first.

f4exb commented:
I think it is also possible to build for Mac with Github actions. It may be
not or less limited. The Windows build is made with Github actions.

On Fri, May 20, 2022, 13:12 srcejon ***@***.***> wrote:

> No - but I've updated the wiki - thanks.
>
> I broke the Mac build when I fixed the Ubuntu build - fix is here:
> master...srcejon:mac_package
> <https://github.com/f4exb/sdrangel/compare/master...srcejon:mac_package>
>
> I'm trying now to see if I can get it to build on appveyor. May need some
> changes to build in under 60 minutes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/f4exb/sdrangel/pull/1249#issuecomment-1132777364>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABPHZP27ITPSRYPLW44R4YDVK5XQVANCNFSM5WGENPFA>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

srcejon commented:
> I think it is also possible to build for Mac with Github actions. It may be not or less limited. The Windows build is made with Github actions.
> […](#)
> On Fri, May 20, 2022, 13:12 srcejon ***@***.***> wrote: No - but I've updated the wiki - thanks. I broke the Mac build when I fixed the Ubuntu build - fix is here: master...srcejon:mac_package <[master...srcejon:mac_package](https://github.com/f4exb/sdrangel/compare/master...srcejon:mac_package)> I'm trying now to see if I can get it to build on appveyor. May need some changes to build in under 60 minutes. — Reply to this email directly, view it on GitHub <[#1249 (comment)](https://github.com/f4exb/sdrangel/pull/1249#issuecomment-1132777364)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABPHZP27ITPSRYPLW44R4YDVK5XQVANCNFSM5WGENPFA> . You are receiving this because you modified the open/close state.Message ID: ***@***.***>

Yes, it looks like it, just not on Mojave (as per the above link) - but the benefits of CI may outweigh one user's request.
